Rubberduck.VBEEditor/ComManagement/TypeLibs/TypeLibs.cs

```diff
@@ -0,0 +1,1801 @@
+            // the guids are dynamic, so we can't use them for detection.
+            if ((TypeKind == TYPEKIND_VBE.TKIND_COCLASS) &&
+                    HasNoContainer() &&
+                    (ImplementedInterfaces.Count == 2) && 
+                    (Name == "Form"))
+            {
+                // we can be 99.999999% sure it IS the runtime generated UserForm base class
+                _isUserFormBaseClass = true;
+            }
+        }
+
+        /// <summary>
+        /// Provides an accessor object for invoking methods on a standard module in a VBA project
+        /// </summary>
+        /// <remarks>caller is responsible for calling ReleaseComObject</remarks>
+        /// <returns>the accessor object</returns>
+        public IDispatch GetStdModAccessor()
+        {
+            if (HasVBEExtensions)
+            {
+                return target_IVBETypeInfo.GetStdModAccessor();
+            }
+            else
+            {
+                throw new ArgumentException("This ITypeInfo is not hosted by the VBE, so does not support GetStdModAccessor");
+            }
+        }
+        
+        /// <summary>
+        /// Executes a procedure inside a standard module in a VBA project
+        /// </summary>
+        /// <param name="name">the name of the procedure to invoke</param>
+        /// <param name="args">arguments to pass to the procedure</param>
+        /// <remarks>the returned object can be a COM object, and the callee is responsible for releasing it appropriately</remarks>
+        /// <returns>an object representing the return value from the procedure, or null if none.</returns>
+        public object StdModExecute(string name, object[] args = null)
+        {
+            if (HasVBEExtensions)
+            {
+                // We search for the dispId using the real type info rather than using staticModule.GetIdsOfNames, 
+                // as we can then also include PRIVATE scoped procedures.
+                var func = Funcs.Find(name, TypeInfoFunc.PROCKIND.PROCKIND_PROC);
+                if (func == null)
+                {
+                    throw new ArgumentException($"StdModExecute failed.  Couldn't find procedure named '{name}'");
+                }
+                
+                var staticModule = GetStdModAccessor();
+                
+                try
+                {
+                    return IDispatchHelper.Invoke(staticModule, func.FuncDesc.memid, IDispatchHelper.InvokeKind.DISPATCH_METHOD, args);
+                }
+                catch (Exception)
+                {
+                    throw;
+                }
+                finally
+                {
+                    Marshal.ReleaseComObject(staticModule);
+                }
+            }
+            else
+            {
+                throw new ArgumentException("This ITypeInfo is not hosted by the VBE, so does not support StdModExecute");
```

Comment:
A common coding technique to keep code readable and reduce the nestings is to invert the `if` statement. Example:

```
if (!HasVBEExtensions)
{
  throw new ArgumentException("This ITypeInfo is not hosted by the VBE, so does not support StdModExecute");
}

// rest of code
```

This way, the messages are now closer to the checking code and control is returned earlier rather than jumping over a large block of code.

Reply:
sure, agreed.